### ISSUE TITLE
Suspect 'pmpro_cron_admin_activity_email' is causing MySQL query stalls on larger WPMS installs

### ISSUE THREAD

**Issue by woodwardmatt** [strangerstudios/paid-memberships-pro#1532]
**Describe the bug**
About a week ago a site of mine with > 1000 subsites migrated over to a new server (key change being MySQL 5.6 to 5.7, but nothing else to my knowledge - this is a managed host). Our site started going offline daily at 0930, On closer inspection MySQL queries were stalling, eventually locking up the entire site and taking it offline.

These queries predominantly (>85%) related to a cron job - I suspect pmpro_cron_admin_activity_email as the query related to "pmpro_visits". I've temporarily disabled the schedule for the cron job to see if this prevents the issue and will update over the next few days to confirm.

**Example Stalled MySQL Query**

| 56295 | <data removed> | localhost | <data removed> | Query   | 3705 | Waiting for table level lock    | UPDATE `wp_options` SE                                                                                                             T `option_value` = 'a:9:{s:5:\"today\";i:517;s:8:\"thisdate\";s:10:\"2020-13-12\";s:5:\"month\";i:18542;s:9:\"thismonth\";s:2:\"                                                                                                             12\";s:7:\"alltime\";i:1247205;s:4:\"week\";i:9188;s:8:\"thisweek\";s:2:\"50\";s:3:\"ytd\";i:656238;s:8:\"thisyear\";s:4:\"2020\                                                                                                             ";}' WHERE `option_name` = 'pmpro_visits' |         0 |             0 |

**WordPress Environment**
WordPress Version | 5.5.1
WP Multisite | Yes
PHP Version: | 7.3.25
MySQL Version | 5.7.26-29

**Comment by dparker1005** [comment]
Hi @woodwardmatt,

Is this an issue that is still happening on your site? In the `Memberships > Settings > Advanced` menu, there is an option to change the frequency of the admin activity email or disable it altogether.

**Comment by woodwardmatt** [comment]
@dparker1005 hey David, I've changed our hosting since experiencing this issue, and with that move the issue disappeared. I think the host had made some changes but wasn't wanting to hold their hands up to them. I changed the cron job frequency at the time, and I was still experiencing issues, so I don't think this was the root cause - just a symptom.

Either way, I think the added option is probably a good thing so the frequency can be managed. 

Thanks for the update :)
